app/code/community/Sendsmaily/Sync/Helper/Data.php

```diff
@@ -21,6 +21,113 @@
 
 class Sendsmaily_Sync_Helper_Data extends Mage_Adminhtml_Helper_Data
 {
+  /**
+   * Validate google reCAPTCHA response.
+   *
+   * @param string $response reCAPTCHA response from form submit.
+   * @return boolean
+   */
+  public function isCaptchaValid($response)
+  {
+    $secretKey =  Mage::getStoreConfig('newsletter/sendsmaily/recaptcha_secret_key');
+    $data = array(
+      'secret' => $secretKey,
+      'response' => $response
+    );
+
+    $curl = new Varien_Http_Adapter_Curl();
+    $curl->setConfig(array('timeout' => 15));
+    $curl->write(
+        Zend_Http_Client::POST,
+        'https://www.google.com/recaptcha/api/siteverify',
+        '1.1',
+        array(),
+        http_build_query($data)
+    );
+
+    $read = $curl->read();
+    $body = json_decode(Zend_Http_Response::extractBody($read), true);
+    $curl->close();
+
+    if (isset($body['success']) && $body['success'] === true) {
+      return true;
+    } else {
+      return false;
+    }
+  }
+
+  /**
+   * Check if its necessary to check CAPTCHA for newsletter form.
+   *
+   * @return boolean
+   */
+  public function shouldCheckCaptcha()
+  {
+    return ((bool) Mage::getStoreConfig('newsletter/sendsmaily/active') === true &&
+        (bool) Mage::getStoreConfig('newsletter/sendsmaily/active_captcha')
+    );
+  }
+
+  /**
+   * Check if newsletter form opt-in collection is enabled.
+   *
+   * @return bool
+   */
+  public function newsletterOptInEnabled()
+  {
+    return ((bool) Mage::getStoreConfig('newsletter/sendsmaily/active') === true &&
+      (bool) Mage::getStoreConfig('newsletter/sendsmaily/active_newsletter_form') == true
+    );
+  }
+
+  /**
+   * Sends newsletter form subscriber information to Smaily
+   *
+   * @param string $email Subscriber email
+   * @param array $data Subscriber data
+   * @return array Response from API call
+   */
+  public function optInSubscribe($email, $extra)
```

Comment:
I think a better name for this method would be `optInSubscriber`. Because "opt-in" and "subscribe" kind of mean the same thing, and method comment says so :)

Reply:
Done in https://github.com/sendsmaily/Sendsmaily-Sync-for-Magento/pull/7/commits/f966d4c267e04b9ae5ca87a4ca6cbe53802c3690